finality-aleph/src/party.rs

```diff
@@ -1,101 +1,87 @@
 use crate::{
-    environment::Environment, network, network::ConsensusNetwork, AuthorityId, AuthorityKeystore,
-    EpochId, NodeId, SpawnHandle,
+    environment::Environment, network, network::ConsensusNetwork, EpochId, NodeId, SpawnHandle,
 };
 
-use futures::channel::mpsc;
-use rush::{Config as ConsensusConfig, Consensus};
+use futures::{channel::mpsc, SinkExt};
+use rush::Consensus;
 use sc_client_api::backend::Backend;
 use sp_consensus::SelectChain;
 use sp_core::{Blake2Hasher, Hasher, H256};
 use sp_runtime::traits::Block;
-use std::{marker::PhantomData, sync::Arc};
 
-pub(crate) struct ConsensusParty<B, N, C, BE, SC>
-where
-    B: Block,
-    N: network::Network<B>,
-    C: crate::ClientForAleph<B, BE> + Send + Sync + 'static,
-    BE: Backend<B> + 'static,
-    SC: SelectChain<B> + 'static,
-{
-    network: ConsensusNetwork<B, H256, N>,
-    client: Arc<C>,
-    select_chain: SC,
-    auth_keystore: AuthorityKeystore,
-    //NOTE: not sure if this phantom is necessary here
-    _phantom: std::marker::PhantomData<BE>,
+pub struct AlephParams<N, C, SC> {
+    pub config: crate::AlephConfig<N, C, SC>,
 }
 
-impl<B, N: network::Network<B>, C, BE, SC> ConsensusParty<B, N, C, BE, SC>
+pub async fn run_consensus_party<B, N, C, BE, SC>(aleph_params: AlephParams<N, C, SC>)
 where
     B: Block,
     N: network::Network<B> + 'static,
     C: crate::ClientForAleph<B, BE> + Send + Sync + 'static,
     BE: Backend<B> + 'static,
     SC: SelectChain<B> + 'static,
 {
-    pub(crate) fn new(
-        client: Arc<C>,
-        network: N,
-        select_chain: SC,
-        auth_keystore: AuthorityKeystore,
-    ) -> Self {
-        let network = ConsensusNetwork::new(network, "/cardinals/aleph/1");
-
-        ConsensusParty {
-            network,
-            client,
-            select_chain,
-            auth_keystore,
-            _phantom: PhantomData,
-        }
-    }
+    // TODO now it runs just a single instance of consensus but later it will
+    // orchestrate managing multiple instances for differents epoch
+    let AlephParams {
+        config:
+            crate::AlephConfig {
+                network,
+                consensus_config,
+                client,
+                select_chain,
+                spawn_handle,
+                auth_keystore,
+                authorities,
+            },
+    } = aleph_params;
+    let network = ConsensusNetwork::new(network, "/cardinals/aleph/1");
+    let spawn_handle: SpawnHandle = spawn_handle.into();
 
-    pub(crate) async fn run(
-        &self,
-        authorities: Vec<AuthorityId>,
-        conf: ConsensusConfig<NodeId>,
-        spawn_handle: SpawnHandle,
-    ) {
-        // TODO now it runs just a single instance of consensus but later it will
-        // orchestrate managing multiple instances for differents epoch
+    let (net_command_tx, net_command_rx) = mpsc::unbounded();
+    let network = network.clone();
```

Comment:
This clone does not seem that useful :)?

Reply:
fixed!